decidim-core/lib/decidim/core/test/shared_examples/comments_examples.rb

```diff
@@ -162,51 +149,67 @@ def visit_commentable_path
+        before do
+          visit resource_path
 
-        expect(page).to have_selector(".add-comment form")
+          expect(page).to have_selector(".add-comment form")
+        end
 
-        page.find(".opinion-toggle--ok").click
+        it "works according to the setting in the commentable" do
+          if commentable.comments_have_alignment?
+            page.find(".opinion-toggle--ok").click
 
-        within ".add-comment form" do
-          fill_in "add-comment-#{commentable.commentable_type}-#{commentable.id}", with: "I am in favor about this!"
-          click_button "Send"
-        end
+            within ".add-comment form" do
+              fill_in "add-comment-#{commentable.commentable_type}-#{commentable.id}", with: "I am in favor about this!"
+              click_button "Send"
+            end
 
-        within "#comments" do
-          expect(page).to have_selector "span.success.label", text: "In favor"
+            within "#comments" do
+              expect(page).to have_selector "span.success.label", text: "In favor"
+            end
+          else
+            expect(page).not_to have_selector(".opinion-toggle--ok")
+          end
         end
       end
     end
 
-    context "when votable option is enabled" do
+    describe "votable option" do
       before do
-        expect_any_instance_of(commentable.class).to receive(:comments_have_votes?).and_return(true)
+        visit resource_path
       end
 
-      it "user can upvote a comment" do
-        visit_commentable_path
-
-        within "#comment_#{comments[0].id}" do
-          expect(page).to have_selector(".comment__votes--up", text: /0/)
-          page.find(".comment__votes--up").click
-          expect(page).to have_selector(".comment__votes--up", text: /1/)
+      context "upvoting a comment" do
+        it "works according to the setting in the commentable" do
+          within "#comment_#{comments[0].id}" do
+            if commentable.comments_have_votes?
+              expect(page).to have_selector(".comment__votes--up", text: /0/)
+              page.find(".comment__votes--up").click
+              expect(page).to have_selector(".comment__votes--up", text: /1/)
+            else
+              expect(page).to_not have_selector(".comment__votes--up", text: /0/)
+            end
+          end
         end
       end
 
-      it "user can downvote a comment" do
-        visit_commentable_path
+      context "downvoting a comment" do
+        before do
+          visit resource_path
+        end
 
-        within "#comment_#{comments[0].id}" do
-          expect(page).to have_selector(".comment__votes--down", text: /0/)
-          page.find(".comment__votes--down").click
-          expect(page).to have_selector(".comment__votes--down", text: /1/)
+        it "works according to the setting in the commentable" do
+          within "#comment_#{comments[0].id}" do
+            if commentable.comments_have_votes?
```

Comment:
Absolutely, that's the idea for a later refactoring. But for now I just followed the same strategy already used in the file (when checking if `commentable.responds_to(:author)`), namely, running conditional expectations according to the characteristics of the commentable.

I had a look at extracting a context but it involved quite a bit of work to do it right. So I preferred to open this PR since it's already an improvement. See also the comment in the relevant commit.